osquery/tables/system/windows/ntfs_acl_permissions.cpp

```diff
@@ -0,0 +1,207 @@
+  case SET_ACCESS:
+    return "Set";
+  case DENY_ACCESS:
+    return "Deny";
+  case REVOKE_ACCESS:
+    return "Revoke";
+  case SET_AUDIT_SUCCESS:
+    return "Set Audit Success";
+  case SET_AUDIT_FAILURE:
+    return "Set Audit Failure";
+  }
+  return "";
+}
+
+// helper function to build access string from permission bit mask
+std::string accessPermsToStr(const unsigned long pmask) {
+  std::vector<std::string> permList;
+  const std::map<unsigned long, std::string> permVals = {
+      {DELETE, "Delete"},
+      {READ_CONTROL, "Read Control"},
+      {WRITE_DAC, "Write DAC"},
+      {WRITE_OWNER, "Write Owner"},
+      {SYNCHRONIZE, "Synchronize"},
+      {STANDARD_RIGHTS_REQUIRED, "Std Rights Required"},
+      {STANDARD_RIGHTS_ALL, "Std Rights All"},
+      {SPECIFIC_RIGHTS_ALL, "Specific Rights All"},
+      {ACCESS_SYSTEM_SECURITY, "Access System Security"},
+      {MAXIMUM_ALLOWED, "Maximum Allowed"},
+      {GENERIC_READ, "Generic Read"},
+      {GENERIC_WRITE, "Generic Write"},
+      {GENERIC_EXECUTE, "Generic Execute"},
+      {GENERIC_ALL, "Generic All"}};
+
+  for (auto const& perm : permVals) {
+    if ((pmask & perm.first) != 0) {
+      permList.push_back(perm.second);
+    }
+  }
+
+  return alg::join(permList, ",");
+}
+
+// helper function to convert inheritance type to string
+std::string inheritanceToStr(const unsigned long i) {
+  switch (i) {
+  case CONTAINER_INHERIT_ACE: // equivalent to SUB_CONTAINERS_ONLY_INHERIT
+    return "Container Inherit Ace";
+  case INHERIT_NO_PROPAGATE: // equivalent to NO_PROPAGATE_INHERIT_ACE
+    return "Inherit No Propagate";
+  case INHERIT_ONLY: // equivalent to INHERIT_ONLY_ACE
+    return "Inherit Only";
+  case OBJECT_INHERIT_ACE: // equivalent to SUB_OBJECTS_ONLY_INHERIT
+    return "Object Inherit Ace";
+  case SUB_CONTAINERS_AND_OBJECTS_INHERIT:
+    return "Sub containers and Objects Inherit";
+  case NO_INHERITANCE:
+    return "No Inheritence";
+  default:
+    return "Other";
+  }
+}
+
+// helper function to get account/group name from trustee
+std::string trusteeToStr(const TRUSTEE& t) {
+  char name[256];
+  char domain[256];
+  unsigned long size = 256; // Max username length for Windows
+  SID_NAME_USE accountType;
+
+  switch (t.TrusteeForm) {
+  case TRUSTEE_IS_SID:
+    // get the name from the SID
+    if (!LookupAccountSid(nullptr,
+                          reinterpret_cast<PSID>(t.ptstrName),
+                          name,
+                          &size,
+                          domain,
+                          &size,
+                          &accountType)) {
+      TLOG << "LookupAccountSid error: " << GetLastError();
+    } else {
+      return name;
+    }
+  case TRUSTEE_IS_NAME:
+    // get the name from ptstrName
+    return t.ptstrName;
+  case TRUSTEE_BAD_FORM:
+    // invalid
+    return "Invalid";
+  case TRUSTEE_IS_OBJECTS_AND_SID:
+    // ptstrName member is a pointer to an OBJECTS_AND_SID struct
+    if (!LookupAccountSid(
```

Comment:
For the MSDN API calls, it's much better to get our hands on the return value instead of having an implicit `if` statement. Let's grab whatever value is here and explicitly check it, and if it's a failure value then report it using a `VLOG(1)` or a `LOG(WARNING)`.